.github/workflows/binaries.yml

```diff
@@ -0,0 +1,72 @@
+name: Binaries
+
+on:
+  push:
+    tags: [v*]
+    branches: [master]
+  pull_request:
+    branches: [master]
+  release:
+    types: [published]
+
+jobs:
+  create-binaries:
+    strategy:
+      matrix:
+        goos: [linux]
+        goarch: [amd64, arm64, ppc64le, riscv64, s390x]
+
+    runs-on: ubuntu-latest
+
+    steps:
+    - name: Checkout
+      uses: actions/checkout@v3
+      with:
+        fetch-depth: 0
+
+    - name: Set up Go
+      uses: actions/setup-go@v4
+      with:
+        go-version: stable  # latest stable version of Go
+
+    - name: Build binary and create archive
+      id: build
+      env:
+        GOOS: ${{ matrix.goos }}
+        GOARCH: ${{ matrix.goarch }}
+        CGO_ENABLED: "0"
+      run: |
+        echo Generating version file
+        go generate ./cmd
+
+        echo Building for $GOOS $GOARCH
+        go build -trimpath -ldflags='-s -w' -o dist/build/pebble ./cmd/pebble
+
+        # Get version via "pebble version" to ensure it matches that exactly
+        PEBBLE_VERSION=$(GOOS=linux GOARCH=amd64 go run ./cmd/pebble version | awk '/client/ { print $2 }')
```

Comment:
We could get the version via GitHub environment variables or git commands, but doing this ensures that it's the exact same version `cmd/mkversion.sh` has produced and embedded in the binary to use in `pebble version` output.

The `pebble version` command currently tries to fetch the server version as well, falling back to `-` if it fails with the timeout of 5s. I might put up a second PR to add a `--client` arg to `pebble version` so to avoid the 5s delay and the `awk`ward parsing.

Reply:
I've created https://github.com/canonical/pebble/pull/208 so we could simplify the above to:

```
PEBBLE_VERSION=$(GOOS=linux GOARCH=amd64 go run ./cmd/pebble version --client)
```